cordova-lib/src/cordova/util.js

```diff
@@ -55,6 +61,8 @@ exports.preProcessOptions = preProcessOptions;
 exports.addModuleProperty = addModuleProperty;
 exports.getOrigWorkingDirectory = getOrigWorkingDirectory;
 exports.fixRelativePath = fixRelativePath;
+exports.getPlatformDetailsFromDir = getPlatformDetailsFromDir;
```

Comment:
"util" is generally a place where code factoring is low because people dump into it. I think this case applies - you should create a new module for the git cloning. Don't care where you put it though. Maybe call it "gitclone.js"